src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -421,7 +448,72 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
+   *   <li>Calls in Starting should be rolled back to NotStarted.</li>
+   *   <li>Calls in Running with no job key should be rolled back to NotStarted.</li>
+   * </ul>
+   *
+   * Calls in Running *with* a job key should be left in Running.  The WorkflowActor is responsible for
+   * resuming the call actors for these calls.
+   */
+  override def prepareForRestart(restartableWorkflow: WorkflowDescriptor)(implicit ec: ExecutionContext): Future[Unit] = {
+
+    lazy val tag = s"Workflow ${restartableWorkflow.id.shortString}:"
+
+    def handleExecutionStatuses(executions: Traversable[Execution]): Future[Unit] = {
+
+      val executionsByStatus = executions.groupBy(_.status) map { case (k, v) => ExecutionStatus.withName(k) -> v }
+      def stringifyExecutions(executions: Traversable[Execution]): String = {
+        executions.toSeq.sortWith((lt, rt) => lt.callFqn < rt.callFqn || (lt.callFqn == rt.callFqn && lt.index < rt.index)).mkString(" ")
+      }
+
+      if (executionsByStatus.contains(ExecutionStatus.Failed)) {
+        val failedExecutions = stringifyExecutions(executionsByStatus.get(ExecutionStatus.Failed).get)
+        Future.failed(new Throwable(s"$tag Cannot restart, found executions in Failed status: " + failedExecutions))
+      }
+      else if (executionsByStatus.contains(ExecutionStatus.Aborted)) {
+        val abortedExecutions = stringifyExecutions(executionsByStatus.get(ExecutionStatus.Aborted).get)
+        Future.failed(new Throwable(s"$tag Cannot restart, found executions in Aborted status: " + abortedExecutions))
+      }
+      else {
+        // Cromwell currently does not persist the types of executions.  Scatters are identified by this magic
+        // "$scatter_" string.
+        val (scatters, nonScatters) = executions partition { _.callFqn.contains(Scatter.FQNIdentifier) }
+        if (scatters.exists(_.status.toExecutionStatus == ExecutionStatus.Starting)) {
+          val startingScatters = stringifyExecutions(scatters)
+          Future.failed(new Throwable(s"$tag Cannot restart, found scatters in Starting status: " + startingScatters))
+        }
+        else {
+          // Scattered calls have multiple executions with the same FQN.  The collector is the execution with no index.
+          // The first element of the partition will be scattered calls (both collectors *and* shards), the second
+          // element is all unscattered calls.
+          val (collectorsAndShards, unscatteredCalls) = nonScatters.groupBy(_.callFqn) partition { case (_, xs) => xs.size > 1  }
+          val (collectors, shards) = collectorsAndShards.values.flatten partition { _.index.toIndex.isEmpty }
+          val calls = unscatteredCalls.values.flatten ++ shards
+
+          val runningCollectors = collectors.filter(_.status.toExecutionStatus == ExecutionStatus.Running)
+          val startingCalls = calls.filter(_.status.toExecutionStatus == ExecutionStatus.Starting)
```

Comment:
Looks like this is not used anymore ? Which means `calls` above probably isn't either. That could simplify some of the `partition`ing above to just be `filter`s I think.